src/eShop.ServiceDefaults/OpenApiOptionsExtensions.cs

```diff
@@ -0,0 +1,229 @@
+                    .Append(when.Date.ToShortDateString())
+                    .Append('.');
+            }
+
+            if (policy.HasLinks)
+            {
+                text.AppendLine();
+
+                var rendered = false;
+
+                foreach (var link in policy.Links.Where(l => l.Type == "text/html"))
+                {
+                    if (!rendered)
+                    {
+                        text.Append("<h4>Links</h4><ul>");
+                        rendered = true;
+                    }
+
+                    text.Append("<li><a href=\"");
+                    text.Append(link.LinkTarget.OriginalString);
+                    text.Append("\">");
+                    text.Append(
+                        StringSegment.IsNullOrEmpty(link.Title)
+                        ? link.LinkTarget.OriginalString
+                        : link.Title.ToString());
+                    text.Append("</a></li>");
+                }
+
+                if (rendered)
+                {
+                    text.Append("</ul>");
+                }
+            }
+        }
+
+        return text.ToString();
+    }
+
+    public static OpenApiOptions ApplySecuritySchemeDefinitions(this OpenApiOptions options)
+    {
+        options.UseTransformer<SecuritySchemeDefinitionsTransformer>();
+        return options;
+    }
+
+    public static OpenApiOptions ApplyAuthorizationChecks(this OpenApiOptions options, string[] scopes)
+    {
+        options.UseOperationTransformer((operation, context, cancellationToken) =>
+        {
+            var metadata = context.Description.ActionDescriptor.EndpointMetadata;
+
+            if (!metadata.OfType<IAuthorizeData>().Any())
+            {
+                return Task.CompletedTask;
+            }
+
+            operation.Responses.TryAdd("401", new OpenApiResponse { Description = "Unauthorized" });
+            operation.Responses.TryAdd("403", new OpenApiResponse { Description = "Forbidden" });
+
+            var oAuthScheme = new OpenApiSecurityScheme
+            {
+                Reference = new OpenApiReference { Type = ReferenceType.SecurityScheme, Id = "oauth2" }
+            };
+
+            operation.Security = new List<OpenApiSecurityRequirement>
+            {
+                new()
+                {
+                    [oAuthScheme] = scopes
+                }
+            };
+
+            return Task.CompletedTask;
+        });
+        return options;
+    }
+
+    public static OpenApiOptions ApplyOperationDefaultValues(this OpenApiOptions options)
+    {
+        options.UseOperationTransformer((operation, context, cancellationToken) =>
+        {
+            var apiDescription = context.Description;
+
+            operation.Deprecated |= apiDescription.IsDeprecated();
+
+            // remove any assumed media types not present in the api description
+            foreach (var responseType in context.Description.SupportedResponseTypes)
```

Comment:
@captainsafia this _might_ not be needed anymore. this was a specific issue with Swashbuckle. I'm not sure if the problem still exists.

If you version by media type (the _truly_ RESTful way 😉), some incorrect assumptions were made. There are multiple possible approaches in an implementation, but the simplest is `application/json; v=#.#`. The API Explorer extensions use the defined `ApiDescription.SupportedRequestFormats`, `ApiDescription.SupportedResponseTypes`, and fans them out per API version. This will approach will take `application/json` and fan it out to `application/json; v=1.0`, `application/json; v=2.0`, and so on. If you're remotely curious how that happens, you can have a peek [here](https://github.com/dotnet/aspnet-api-versioning/blob/3fc071913dcded23eeb5ebe55bca44f3828488bf/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc.ApiExplorer/ApiVersionParameterDescriptionContext.cs#L202).

This can be problematic because if _assuming_ is allowed (e.g. `AssumeDefaultVersionWhenUnspecified = true`), then `application/json` will work. If it's not, which is the default, then it will not work, but `application/json` is listed as a valid media type. While it's technical possible, there is no support for yielding just `application/json` if matching an implicit version is allowed. If the default version were `1.0`, then `application/json` and `application/json; v=1.0` are equivalent anyway. This is also true for an API _version-neutral_ endpoint.

If this behavior doesn't exist anymore, then TL;DR - I get it. This behavior does still exist, it should **only** happen if there are no other media types defined (IMO). I don't believe that behavior needs to exist here though.

Reply:
Yeah, upon further inspection, we don't need to handle setting default values, fixing requiredness on parameters, or fixing parameters on descriptions.